pkg/labels/labels.go

```diff
@@ -139,6 +139,10 @@ const (
 
 	// LabelSourceReservedKeyPrefix is the prefix of a reserved label
 	LabelSourceReservedKeyPrefix = LabelSourceReserved + "."
+
+	// LabelSourceCiliumGenerated is for labels auto-generated by cilium without
+	// user input
+	LabelSourceCiliumGenerated = "cilium-generated"
```

Comment:
Are there any considerations we may need to worry about for whether someone could mess with the policy engine by manually inserting rules containing labels prefixed with this source?

Reply:
I know we have such logic for endpoint API, not sure about policies. But this seems like a natural fit somewhere around the policy sanitization logic.